src/datachain/lib/signal_schema.py

```diff
@@ -550,7 +550,7 @@ def _set_file_stream(
     ) -> None:
         if isinstance(obj, File):
             obj._set_stream(catalog, caching_enabled=cache)
-        for field, finfo in obj.model_fields.items():
+        for field, finfo in obj.__class__.model_fields.items():
```

Comment:
While `obj.__class__` reads nicely, I reckon that `type(obj)` would have been slightly more Pythonic:

```suggestion
        for field, finfo in type(obj).model_fields.items():
```

See e.g. [pydantic@d156ba0 → tests/benchmarks/basemodel_eq_performance.py:58](https://github.com/pydantic/pydantic/blob/d156ba08c140ee1e2b931120cb150080843476fe/tests/benchmarks/basemodel_eq_performance.py#L58) 💅🏼

Reply:
less Pythonic more ... Ruby? :) (Python has to be __init__(self) something .. :) ) 

I like it though, will update it

Reply:
I requested GitHub Copilot's review after mine, and it agrees; not sure if it's because it parsed my comments, though. 📋 

* https://github.com/iterative/datachain/pull/1242#discussion_r2208709506